url29ecd5d728_geodb_db3dcore_tgz-pJ8-de_uos_igf_db3d_dbms_model3d_SegmentNetBuilderJ8/src/de/uos/igf/db3d/dbms/geom/Tetrahedron3D.java

```diff
@@ -0,0 +1,2623 @@
+	 * @throws IllegalStateException
+	 *             - if the result of the intersection of this with a Plane3D is
+	 *             not a 3D simplex. The exception originates in the method
+	 *             intersection(Plane3D, ScalarOperator) of this class.
+	 * @throws IllegalArgumentException
+	 *             - if index of a triangle point is not 0, 1 or 2. The
+	 *             exception originates in the method getPoint(int) of the class
+	 *             Triangle3D.
+	 * @throws ArithmeticException
+	 *             - if norm equals zero in epsilon range. This exception
+	 *             originates in the method normalize(ScalarOperator) of the
+	 *             class Vector3D.
+	 */
+	public int intersectsInt(Plane3D plane, ScalarOperator sop) { // Dag
+		// evaluation of intersection(plane) return
+		SimpleGeoObj result = this.intersection(plane, sop);
+		// Here an IllegalStateException can be thrown signaling problems with
+		// the dimensions of the wireframe.
+
+		if (result == null)
+			return -1;
+
+		switch (result.getType()) {
+		case SimpleGeoObj.POINT3D:
+			return 0;
+		case SimpleGeoObj.SEGMENT3D:
+			return 1;
+		default:
+			// result must be SimpleGeoObj.TRIANGLE3D or WIREFRAME3D with dim=2
+			return 2;
+		}
+	}
+
+	/**
+	 * Tests whether this intersects the given line and returns a value for the
+	 * dimension of the resulting object. Possible results are: [ -1 for no
+	 * intersection, 0 ->0D (point) or 1 ->1D (segment) ].
+	 * 
+	 * @param line
+	 *            Line3D for test
+	 * @return int - dimension of result.
+	 * @throws IllegalStateException
+	 *             - if the intersectsInt(Line3D line, ScalarOperator sop)
+	 *             method of the class Line3D (which computes the intersection
+	 *             of two lines) called by this method returns a value that is
+	 *             not -2, -1, 0 or 1.
+	 * @throws IllegalStateException
+	 *             - if the result of the intersection of this with a Line3D is
+	 *             not Segment3D, Point3D or null. The exception originates in
+	 *             the method intersection (Line3D, ScalarOperator) of this
+	 *             class.
+	 * @throws IllegalArgumentException
+	 *             - if index of a triangle point is not 0, 1 or 2. The
+	 *             exception originates in the method getPoint(int) of the class
+	 *             Triangle3D.
+	 * @throws ArithmeticException
+	 *             - if norm equals zero in epsilon range. This exception
+	 *             originates in the method normalize(ScalarOperator) of the
+	 *             class Vector3D.
+	 */
+	public int intersectsInt(Line3D line, ScalarOperator sop) { // Dag
+		// evaluation of intersection(line) return
+		SimpleGeoObj result = this.intersection(line, sop);
+
+		if (result == null)
+			return -1;
+
+		switch (result.getType()) {
+		case SimpleGeoObj.SEGMENT3D:
+			return 1;
+		default: // result must be SimpleGeoObj.POINT3D:
+			return 0;
+		}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 1079:1089, 1121:1131.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.